File: launchdarkly/python/src/CreateFlagLinkExample.py
import json
from datetime import date, datetime
from pprint import pprint

from launchdarkly_client import ApiClient, ApiException, Configuration, api, models

configuration = Configuration(
    api_key={"ApiKey": "YOUR_API_KEY"},
)

with ApiClient(configuration) as api_client:
    flag_link_post = models.FlagLinkPost(
        key="flag-link-key-123abc",
        deepLink="https://example.com/archives/123123123",
        title="Example link title",
        description="Example link description",
    )

    try:
        response = api.FlagLinksBetaApi(api_client).create_flag_link(
            project_key=None,
            feature_flag_key=None,
            flag_link_post=flag_link_post,
        )

        pprint(response)
    except ApiException as e:
        print("Exception when calling FlagLinksBeta#create_flag_link: %s\n" % e)

File: launchdarkly/ruby/src/CreateFlagLinkExample.rb
require "json"
require "launchdarkly_client"

LaunchDarklyClient.configure do |config|
    config.api_key["Authorization"] = "YOUR_API_KEY"
end

flag_link_post = LaunchDarklyClient::FlagLinkPost.new
flag_link_post.key = "flag-link-key-123abc"
flag_link_post.deep_link = "https://example.com/archives/123123123"
flag_link_post.title = "Example link title"
flag_link_post.description = "Example link description"

begin
    response = LaunchDarklyClient::FlagLinksBetaApi.new.create_flag_link(
        "projectKey_string", # project_key
        "featureFlagKey_string", # feature_flag_key
        flag_link_post,
    )

    p response
rescue LaunchDarklyClient::ApiError => e
    puts "Exception when calling FlagLinksBetaApi#create_flag_link: #{e}"
end
